integration_tests/testsuite/deploy_app.py

```diff
@@ -102,37 +157,56 @@ def deploy_app(base_image, builder_image, appdir, yaml):
         if builder_image:
             _set_builder_image(builder_image)
 
-        deployed_version = test_util.generate_version()
+        if environment == 'GAE':
+            return deploy_app_gae(yaml)
+        elif environment == 'GKE':
+            return deploy_app_gke(yaml)
 
-        # TODO: once sdk driver is published, use it here
-        deploy_command = ['gcloud', 'app', 'deploy', '--no-promote',
-                          '--version', deployed_version, '-q']
-        if yaml:
-            logging.info(yaml)
-            deploy_command.append(yaml)
-
-        subprocess.check_output(deploy_command)
-
-        return deployed_version
     except subprocess.CalledProcessError as cpe:
         logging.error('Error encountered when deploying application! %s',
                       cpe.output)
-        sys.exit(1)
-
+        raise cpe
+    except Exception as e:
+        logging.error('Error encountered when deploying application! %s', e)
+        raise e
     finally:
-        _cleanup(appdir)
+        _cleanup_files(appdir)
         os.chdir(owd)
 
 
 @retry(wait_fixed=4000, stop_max_attempt_number=8)
-def stop_app(deployed_version):
-    logging.debug('Removing application version %s', deployed_version)
-    try:
-        delete_command = ['gcloud', 'app', 'services', 'delete', 'default',
-                          '--version', deployed_version, '-q']
-
-        subprocess.check_output(delete_command)
-    except subprocess.CalledProcessError as cpe:
-        logging.error('Error encountered when deleting app version! %s',
-                      cpe.output)
-        raise cpe
+def stop_app(version_or_namespace, environment):
```

Comment:
Could you make this two separate functions, instead of using a string arg to switch the behavior?